src/test/java/org/jenkinsci/plugins/envinject/BuildCauseRetrieverTest.java

```diff
@@ -139,6 +142,25 @@ public String getShortDescription() {
         assertThat(build, withCause(ROOT_BUILD_CAUSE, ""));
     }
 
+    @SuppressWarnings("deprecation")
+    @Test
+    public void shouldWriteInfoUserCause() throws Exception {
+        UserCause cause = Cause.UserCause.class.newInstance();
```

Comment:
when you create cause in such way, username and user id is null. Please specify it explicitly